Frontend/src/components/Patients.jsx

```diff
@@ -1,167 +1,226 @@
-import React, { useState } from 'react';
-import { motion } from 'framer-motion';
-import { FaUserPlus, FaSearch, FaEdit, FaTrash } from 'react-icons/fa';
+import React, { useEffect, useState } from "react";
+import { motion } from "framer-motion";
+import { FaUserPlus, FaSearch, FaEdit, FaTrash } from "react-icons/fa";
+import axios from "axios";
 
 const Patients = () => {
-    const [patients, setPatients] = useState([
-        { id: 1, name: "John Doe", age: 35, gender: "Male", contact: "123-456-7890" },
-        { id: 2, name: "Jane Smith", age: 28, gender: "Female", contact: "987-654-3210" },
-    ]);
-    const [searchTerm, setSearchTerm] = useState('');
-    const [showForm, setShowForm] = useState(false);
-    const [newPatient, setNewPatient] = useState({ name: '', age: '', gender: '', contact: '' });
+  const [patients, setPatients] = useState([
+    {
+      id: 1,
+      name: "John Doe",
+      age: 35,
+      gender: "Male",
+      contact: "123-456-7890",
+    },
+    {
+      id: 2,
+      name: "Jane Smith",
+      age: 28,
+      gender: "Female",
+      contact: "987-654-3210",
+    },
+  ]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [showForm, setShowForm] = useState(false);
+  const [newPatient, setNewPatient] = useState({
+    name: "",
+    age: "",
+    gender: "",
+    mobile: "",
+    email: "",
+    password: "",
+  });
 
-    const handleInputChange = (e) => {
-        setNewPatient({ ...newPatient, [e.target.name]: e.target.value });
-    };
+  const handleInputChange = (e) => {
+    setNewPatient({ ...newPatient, [e.target.name]: e.target.value });
+  };
 
-    const validateForm = () => {
-        const nameRegex = /^[A-Za-z][A-Za-z0-9 ]{3,}$/;
-        const contactRegex = /^[0-9]{3}-[0-9]{3}-[0-9]{4}$/;
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
```

Comment:
Also alert for invalid formatting for mobile number.
alert('Invalid contact number. It should follow the format XXX-XXX-XXXX.'